src/state/swap/hooks.tsx

```diff
@@ -116,33 +115,14 @@ export function useDerivedSwapInfo(state: SwapState, chainId: ChainId | undefine
     [inputCurrency, isExactIn, outputCurrency, typedValue]
   )
 
-  let trade = useDebouncedTrade(
+  const trade = useDebouncedTrade(
     isExactIn ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT,
     parsedAmount,
     (isExactIn ? outputCurrency : inputCurrency) ?? undefined,
     undefined,
     account
   )
 
-  const nextPreviouslyInvalid = (() => {
```

Comment:
q: why do we no longer need this previouslyInvalid check?

Reply:
this was adding rendering latency, so i dropped it. i did not understand what it added.

Reply:
not latency, jank. state doesn't render until one frame later, and this uses `useState`.